portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -24,9 +27,24 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+  ArrayList<String> messages;
+
+  public DataServlet() {
+    messages = new ArrayList<>();
+    messages.add("Hello, world!");
+    messages.add("Happy coding!");
+    messages.add("LGTM");
+  }
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello, Joshua!</h1>");
+    response.getWriter().println(convertToJson(messages));
+
+  }
+
+  private String convertToJson(ArrayList<String> messages) {
```

Comment:
Here too, use the List interface.

In general: Unless you're constructing a concrete type, where you need to know exactly what you're constructing (e.g. `new ArrayList<>(...)`), you generally want to pass around a general interface.

Reply:
Same here as above.